Add neo4j to the universe

unterstein opened the pull request:
Additional documentation can be found here:
https://github.com/neo4j-contrib/neo4j-dcos

PR against example here:
https://github.com/dcos/examples/pull/31

ryadav88 commented on repo/packages/N/neo4j-proxy/0/config.json:
```diff
@@ -0,0 +1,28 @@
+        "auth-password": {
+          "description": "Password for authentication",
+          "type": "string",
+          "default": "dcos"
```
the proxy UI says the default username/password is neo4j/neo4j. which one is correct?

unterstein replied:
This configuration. This point is very confusing, because it is semantically like `If you don`t have configured authentication information, this is your login`. @jexp of neo4j opened an ticket to change this in something like my quote.

ryadav88 commented on repo/packages/N/neo4j-proxy/0/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "packagingVersion": "3.0",
+  "name": "neo4j-proxy",
+  "version": "1.0.0",
```
`version` should be `neo4jProxyVersion-packagingVersion`. 
Ex: if neo4jProxyVersion is 1.0.0 then `version` would be `1.0.0-0.0.1`

ryadav88 commented on repo/packages/N/neo4j-proxy/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "version": "1.0.0",
+  "minDcosReleaseVersion" : "1.8",
+  "scm": "https://github.com/neo4j-contrib/neo4j-dcos",
+  "maintainer": "feedback@neo4j.com",
```
Is Neo4j supporting this package?

unterstein replied:
Yes

ryadav88 commented on repo/packages/N/neo4j-proxy/0/marathon.json.mustache:
```diff
@@ -0,0 +1,39 @@
+  "container":{  
+    "type":"DOCKER",
+    "docker":{  
+      "image":"neo4j/neo4j-dcos-proxy:1.0.0",
```
assets.container.docker.dcos-neo4j

ryadav88 commented on repo/packages/N/neo4j-replica/0/marathon.json.mustache:
```diff
@@ -0,0 +1,47 @@
+  "container":{  
+    "type":"DOCKER",
+    "docker":{  
+      "image":"neo4j/neo4j-dcos:1.0.0-3.1-RC1",
```
please change this to get the values from resource.json

ryadav88 commented on repo/packages/N/neo4j-replica/0/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "packagingVersion": "3.0",
+  "name": "neo4j-replica",
+  "version": "1.0.0-3.1.0-RC1",
```
see comment in `proxy` for `version`

ryadav88 commented on repo/packages/N/neo4j-replica/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "version": "1.0.0-3.1.0-RC1",
+  "minDcosReleaseVersion" : "1.8",
+  "scm": "https://github.com/neo4j-contrib/neo4j-dcos",
+  "maintainer": "feedback@neo4j.com",
```
see comment in `proxy` for `maintainer`

ryadav88 commented on repo/packages/N/neo4j-replica/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "framework": false,
+  "description": "Neo4j is an open source Graph Database. It allows you to efficiently manage highly connected data. This DC/OS package is about the Neo4j read replica servers, read replicas act like caches for the data that the core servers safeguard. You can access the cluster via officially supported drivers from most languages. Installation documentation can be found at https://github.com/dco ...
+  "tags": ["neo4j","cypher","nosql","database","graph","acid","graphdb","data"],
+  "preInstallNotes": "The default configuration requires at least 2 nodes having 2 CPU, 4GB of memory and 8GB of persistent disk storage in total. This DC/OS Service is currently EXPERIMENTAL. There may be bugs, incomplete features, incorrect documentation, or other discrepancies. Default credentials are neo4j/dcos. Please change them on first use.",
```
`This DC/OS Service is currently EXPERIMENTAL. There may be bugs, incomplete features, incorrect documentation, or other discrepancies.` should be at the beginning of `preInstall` notes

ryadav88 commented on repo/packages/N/neo4j-replica/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "postUninstallNotes": "Neo4j has been uninstalled.",
+  "licenses": [
+    {
+      "name": "GPL v3 license or Neo4j Commercial/Evaluation/Education License",
```
are you sure we should use education license here? If yes, please point that out in the `description` so customers don't use it.

jexp replied:
According to @joerg84 the general approach is bring-your-own-license. So any of those would be a valid license to be used.
And it should say AGPL instead of GPL.

ryadav88 replied:
thanks @jexp. 
@unterstein please update it and we're all set. Thanks

ryadav88 commented on repo/packages/N/neo4j/0/marathon.json.mustache:
```diff
@@ -0,0 +1,48 @@
+  "container":{  
+    "type":"DOCKER",
+    "docker":{  
+      "image":"neo4j/neo4j-dcos:1.0.0-3.1-RC1",
```
see comment from `proxy`

ryadav88 commented on repo/packages/N/neo4j/0/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "packagingVersion": "3.0",
+  "name": "neo4j",
+  "version": "1.0.0-3.1.0-RC1",
```
see comment from `proxy`

ryadav88 commented on repo/packages/N/neo4j/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "version": "1.0.0-3.1.0-RC1",
+  "minDcosReleaseVersion" : "1.8",
+  "scm": "https://github.com/neo4j-contrib/neo4j-dcos",
+  "maintainer": "feedback@neo4j.com",
```
see comment from `proxy`

ryadav88 commented on repo/packages/N/neo4j/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "framework": false,
+  "description": "Neo4j is an open source Graph Database. It allows you to efficiently manage highly connected data. This DC/OS package is about the Neo4j core cluster servers, core servers' main responsibility is to safeguard data. You can access the cluster via officially supported drivers from most languages. Installation documentation can be found at https://github.com/dcos/examples/tree/mast ...
+  "tags": ["neo4j","cypher","nosql","database","graph","acid","graphdb","data"],
+  "preInstallNotes": "The default configuration requires at least 3 nodes having 3 CPU, 6GB of memory and 12GB of persistent disk storage in total. This DC/OS Service is currently EXPERIMENTAL. There may be bugs, incomplete features, incorrect documentation, or other discrepancies. Default credentials are neo4j/dcos. Please change them on first use.",
```
please move to beginning : `This DC/OS Service is currently EXPERIMENTAL. There may be bugs, incomplete features, incorrect documentation, or other discrepancies.`

unterstein commented:
Wow, thanks for that great feedback! :)

I will address your points tomorrow. I will also correct the resident volume behavior within this the marathon definitions.

unterstein commented:
Addressed all feedback, except the licensing. My partner from neo4j @jexp provided this information and should answer this question.

Are the other changes ok for you?

Thanks
Johannes

ryadav88 commented:
@unterstein thanks for the changes. Yes, LGTM 👍 
Ready to merge once we get a reply on the license. 

If you need this merged with the current license, please add a note in `preInstallNotes` for all 3 highlighting the fact that it has an education license only and should not be used in production.
